src/test/java/org/prebid/server/bidder/adquery/AdQueryBidderTest.java

```diff
@@ -0,0 +1,417 @@
+
+        // then
+        assertThat(result.getValue())
+                .hasSize(1)
+                .extracting(BidderBid::getBidCurrency)
+                .containsExactly("PLN");
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    private static BidRequest givenBidRequest(UnaryOperator<BidRequest.BidRequestBuilder> requestCustomizer,
+                                              Imp... imps) {
+        return requestCustomizer.apply(BidRequest.builder()
+                .id("22e26bd9a702bc")
+                .imp(List.of(imps))
+                .user(User.builder().id("user-id").build())
+                .device(Device.builder()
+                        .ip("31.21.32.42")
+                        .ipv6("131.123.632.32")
+                        .ua("ua-test-value")
+                        .build())
+                .site(Site.builder().page("https://www.unknowsite.com")
+                        .build())).build();
+    }
+
+    private static Imp givenImp(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return impCustomizer.apply(Imp.builder()
+                        .id("1")
+                        .tagid("test-banner-imp-id")
+                        .banner(Banner.builder()
+                                .format(List.of(
+                                        Format.builder().w(320).h(100).build(),
+                                        Format.builder().w(300).h(250).build()))
+                                .build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpAdQuery.of("pl1", "6d93f2a0e5f0fe2cc3a6e9e3ade964b43b07f897")))))
+                .build();
+    }
+
+    private static BidderCall<AdQueryRequest> givenHttpCall(String body) {
+        return BidderCall.succeededHttp(null, HttpResponse.of(200, null, body), null);
+    }
+
+    private static BidderCall<AdQueryRequest> givenHttpCall(
+            UnaryOperator<AdQueryDataResponse.AdQueryDataResponseBuilder> adQueryResponse)
+            throws JsonProcessingException {
+        final String body = mapper.writeValueAsString(
+                AdQueryResponse.of(
+                        adQueryResponse.apply(
+                                        AdQueryDataResponse.builder()
+                                                .requestId("abs")
+                                                .cpm(BigDecimal.valueOf(321))
+                                                .creationId("312")
+                                                .adqLib("AnyLib")
+                                                .currency("UAH")
+                                                .tag("AnyTag")
+                                                .adDomains(singletonList("any"))
+                                                .adQueryMediaType(
+                                                        AdQueryMediaType.of(BidType.banner, 120, 320)))
+                                .build()));
```

Comment:
too deep intendations (it's better to have not more than 8 spaces), make sense to extract something to a separate variable